pkg/controllers/leaderworkerset_controller.go

```diff
@@ -640,6 +669,12 @@ func exclusiveConditionTypes(condition1 metav1.Condition, condition2 metav1.Cond
 		(condition1.Type == string(leaderworkerset.LeaderWorkerSetProgressing) && condition2.Type == string(leaderworkerset.LeaderWorkerSetAvailable)) {
 		return true
 	}
+
+	if (condition1.Type == string(leaderworkerset.LeaderWorkerSetAvailable) && condition2.Type == string(leaderworkerset.LeaderWorkerSetUpgradeInProgress)) ||
```

Comment:
e2e test case is a bit tricky since the `rollingUpdate` can finish before we can check if `UpgradeInProgress` is set to true. I added checks in existing `rollingUpdate` integration tests since those control when the `rollingUpdate` is in progress, and that's where the tests for `Progressing` condition are made as well.

Reply:
we could make the upgrade get stuck like specifying a wrong image, so that the upgrade would stop right?

Reply:
Yeah, but not sure the benefits of doing that as well as having integration tests, the integration test covers the same usage, but without needing to give a bad image.

Reply:
the benefits is the same for adding the e2e test right? only when we really see the condition shows up in the e2e test correctly, so could make sure this works and won't be broken in future by other changes.